File: C_Programming/Competitive/Assignment41_2.c
/*
    Write a recursive program which accept number from user and return
    summation of its digits.
    Input : 879
    Output : 24
*/

#include<stdio.h>

int Sum(int iValue)
{
    static int iSum = 0 ;
    int iDigit = 0 ;

    if(iValue != 0)
    {
        iDigit = iValue % 10 ;
        iSum = iSum + iDigit ;
        
        iValue = iValue / 10 ;

        Sum(iValue);
    }
    return iSum;
}

int main()
{
    int iValue = 0 ;
    int iRet = 0 ;
    printf("Enter number : ");
    scanf("%d",&iValue);

    iRet = Sum(iValue);

    printf("Result is : %d\n",iRet);

    return 0 ;
}

File: Java_Programming/Competitive/Assignment41_2.java

import java.util.Scanner;

class Assignment41_2
{
    static int iSum = 0 ;

//--------------------------------------------------------------------//
// Function Name : Sum
// Description   : Calculates the sum of digits recursively.
// Input         : Integer value
// Output        : Sum of digits
// Author        : Pranav Avinash Narkhede
// Date          : 23-07-2026
//--------------------------------------------------------------------//
    static int Sum(int iValue )
    {
        if(iValue != 0)
        {   
            iSum = iSum + (iValue % 10) ;
            Sum(iValue/10);
        }

        return iSum;
    }

//--------------------------------------------------------------------//
// Application to calculate the sum of digits using recursion.
//--------------------------------------------------------------------//
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0 ;
        int iRet = 0 ;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        iRet = Sum(iValue );

        System.out.println("Summation of digits is : "+iRet);
    }
}